Source/Scene/Billboard.js

```diff
@@ -762,6 +806,93 @@ define([
         return this._pickId;
     };
 
+    var scratchRay = new Ray();
+    var scratchPosition = new Cartesian3();
+    var scratchCartographic = new Cartographic();
+
+    Billboard._clampPosition = function(object, mode, projection, markObjectDirty) {
+        var modeChanged = mode !== SceneMode.MORPHING && mode !== object._mode;
+        if (object._newTile !== object._currentTile || object._positionChanged || modeChanged) {
+            if (mode === SceneMode.SCENE3D) {
+                Cartesian3.clone(Cartesian3.ZERO, scratchRay.origin);
+                Cartesian3.normalize(object.position, scratchRay.direction);
+            } else {
+                var ellipsoid = projection.ellipsoid;
+                ellipsoid.cartesianToCartographic(object.position, scratchCartographic);
+                scratchCartographic.height = -1000.0; // TODO: get minimum height of entire terrain set
+                projection.project(scratchCartographic, scratchPosition);
+                Cartesian3.fromElements(scratchPosition.z, scratchPosition.x, scratchPosition.y, scratchPosition);
+                Cartesian3.clone(scratchPosition, scratchRay.origin);
+                Cartesian3.clone(Cartesian3.UNIT_X, scratchRay.direction);
+            }
+
+            var position = object._newTile.data.pick(scratchRay, mode, projection, false, scratchPosition);
+            if (defined(position)) {
+                object._clampedPosition = Cartesian3.clone(position, object._clampedPosition);
+            }
+
+            object._positionChanged = false;
+            object._mode = mode;
+            object._currentTile = object._newTile;
+        }
+    };
+
+    function createCallback(collection, object) {
+        return function (tile) {
+            object._newTile = tile;
+            var clampList = collection._clampToTerrainList;
+            if (clampList.indexOf(object) === -1) {
+                clampList.push(object);
+            }
+        };
+    }
+
+    Billboard.prototype._updateClamping = function() {
+        Billboard._updateClamping(this._billboardCollection, this);
+    };
+
+    Billboard._updateClamping = function(collection, object) {
+        var globe = collection._globe;
+        if (!defined(globe)) {
+            if (defined(object._clampToGround)) {
+                throw new DeveloperError('Clamping a billboard to the ground is not supported.');
```

Comment:
I don't have this all worked yet, but the scene knows the globe and the scene passes a frameState to primitives so perhaps the frameState could provide something to subscribe to?  Otherwise, we could require passing the scene to the billboard collection during construction, which would allow the user to reach into fewer objects and allow us to support the unlikely case when the globe changes (I wouldn't bother to start though).